src/main/java/com/google/cloud/teleport/v2/bigquery-gcs/src/main/java/com/google/cloud/teleport/v2/bigquerygcs/BigQueryToParquet.java

```diff
@@ -0,0 +1,261 @@
+                            .setTableReference(storageTableRef);
+            try {
+                return client.createReadSession(builder.build());
+            } catch (InvalidArgumentException iae) {
+                LOG.error("Error creating ReadSession: " + iae.getMessage());
+                throw iae;
+            }
+        }
+    }
+
+    /**
+     * The {@link BigQueryToParquetOptions} class provides the custom execution options passed by the executor at the
+     * command-line.
+     */
+    public interface BigQueryToParquetOptions extends PipelineOptions {
+        @Description("BigQuery table to export from in the form <project>:<dataset>.<table>")
+        @Required
+        String getTableRef();
+
+        void setTableRef(String tableRef);
+
+        @Description("GCS bucket to export BigQuery table data to (e.g. gs://mybucket/folder/).")
+        @Required
+        String getBucket();
+
+        void setBucket(String bucket);
+
+        @Description("Optional: Number of shards for output file.")
+        @Default.Integer(0)
+        Integer getNumShards();
+
+        void setNumShards(Integer numShards);
+
+        @Description("Optional: Comma separated list of fields to select from the table.")
+        String getFields();
+
+        void setFields(String fields);
+    }
+
+    /**
+     * The {@link getTableSchema} method gets Avro schema for table using BigQueryStorage API.
+     * @param session ReadSession that contains schema for table, filtered by fields if any.
+     * @return avroSchema Avro schema for table. If fields are provided then schema will only contain those fields.
+     */
+    private static Schema getTableSchema(ReadSession session) {
+        Schema avroSchema;
+
+        avroSchema = new Schema.Parser().parse(session.getAvroSchema().getSchema());
+        LOG.info("Schema for export is: " + avroSchema.toString());
+
+        return avroSchema;
+    }
+
+    public static void main(String[] args) {
+        BigQueryToParquetOptions options =
+                PipelineOptionsFactory.fromArgs(args).withValidation()
+                        .as(BigQueryToParquetOptions.class);
+        run(options);
+    }
+
+    /**
+     * Runs the pipeline with the supplied options.
+     *
+     * @param options The execution parameters to the pipeline.
+     * @return The result of the pipeline execution.
+     */
+    public static PipelineResult run(BigQueryToParquetOptions options) {
+
+        Pipeline pipeline = Pipeline.create(options);
+
+        TableReadOptions.Builder builder = TableReadOptions.newBuilder();
+
+        if (options.getFields() != null) {
+            builder.addAllSelectedFields(Arrays.asList(options.getFields().split(",\\s*")));
+        }
+
+        TableReadOptions tableReadOptions = builder.build();
+        ReadSessionFactory readSessionFactory = new ReadSessionFactory();
+        BigQueryStorageClient client = new BigQueryStorageClientFactory().create();
```

Comment:
Same comment here regarding the static method.

Reply:
Done.